### PR TITLE
ICU-22877 revised "Region code" (formerly "Country code") section

### PR DESCRIPTION
**Pull request by cjchapman**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22877
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- ~~[ ] Tests included, if applicable~~ (not applicable)
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by eggrobin — DISMISSED** [review]

File: docs/userguide/locale/index.md [hunk 1]
```diff
@@ -97,17 +97,20 @@ The optional four-letter script code follows the language code. If specified, it
...
 example, Spanish is spoken in many countries in Central and South America but
 the currencies are different in each country. To allow for these differences
-among specific geographical, political, or cultural regions, locales are
-specified by two-letter, uppercase codes. For example, "ES" represents Spain and
-"MX" represents Mexico. The two letter country code uses the
-[ISO-3166](https://www.iso.org/iso-3166-country-codes.html) standard.
-
-Java supports two letter country codes that uses ISO-3166 and UN M.49 code.
+among specific geographical, political, or cultural regions, locales follow
+the [BCP 47](https://www.rfc-editor.org/rfc/bcp/bcp47.txt) convention of specifying
```
Comment (markusicu) [inline docs/userguide/locale/index.md:106]:
```suggestion
the [BCP 47](https://www.rfc-editor.org/info/bcp47) convention of specifying
```
--> nicer page, and easy access to the HTML versions

File: docs/userguide/locale/index.md [hunk 2]
```diff
@@ -97,17 +97,20 @@ The optional four-letter script code follows the language code. If specified, it
...
+regions by using two-letter-uppercase
+[ISO-3166](https://www.iso.org/iso-3166-country-codes.html) or three-digit
+[UN M.49](https://unstats.un.org/unsd/methodology/m49/) codes. For example,
+"ES" represents Spain, "MX" represents Mexico, and "419" represents Latin America
+and the Caribbean. For more information, please see the
+[unicode_region_subtag](https://www.unicode.org/reports/tr35/#unicode_region_subtag_validity)
+section of the [Locale Data Markup Language](http://www.unicode.org/reports/tr35/).
```
Comment (markusicu) [inline docs/userguide/locale/index.md:113]:
```suggestion
section of the [Locale Data Markup Language](https://www.unicode.org/reports/tr35/).
```

**Review by markusicu — CHANGES_REQUESTED** [review]

File: docs/userguide/locale/index.md (hunk 1 above)
Comment (cjchapman) [reply docs/userguide/locale/index.md]: done

File: docs/userguide/locale/index.md (hunk 2 above)
Comment (cjchapman) [reply docs/userguide/locale/index.md]: done

**Comment by cjchapman** [comment]
I amended my last commit and did a force push so this PR still "includes exactly 1 commit" per the `single-commit` check.

Is that the correct procedure when changes are requested, or is it acceptable to have subsequent commits to make the requested changes?

**Comment by markusicu** [comment]
> I amended my last commit and did a force push so this PR still "includes exactly 1 commit" per the `single-commit` check.
> 
> Is that the correct procedure when changes are requested, or is it acceptable to have subsequent commits to make the requested changes?

It is usually better for tracking comments in GitHub PRs to make separate commits, and squash after approval. The approval should stay alive across squashing. (It won't stay alive if you also rebase, so if you don't have merge conflicts, then just squash.)

**Review by markusicu — APPROVED** [review]

**Comment by cjchapman** [comment]
Looks like squash & merge is not enabled, so I guess I'd have to squash locally and then force-push?
Screenshot 2024-09-05 at 3 22 05 PM

**Comment by markusicu** [comment]
You can squash locally and force-push, or you can use the web UI tool from the link next to the failing check.

We don't allow squash-and-merge because that would circumvent the checks that ensure cross-linking between pull requests/commits and tickets.

**Pull request merged**